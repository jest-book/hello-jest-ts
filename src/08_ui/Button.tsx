import { useState } from 'react'
import './button.css';

interface ButtonProps {
  primary?: boolean;
}

export const Button = ({ primary = true, ...props }: ButtonProps) => {
  const [isToggleOn, setIsToggleOn] = useState(true)

  const className = primary ? "primary" : "secondary"

  return (
    <button className={className} onClick={() => setIsToggleOn(!isToggleOn)}>
      {isToggleOn ? 'ON' : 'OFF'}
    </button >
  )
}
