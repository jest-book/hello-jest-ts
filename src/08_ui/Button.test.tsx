/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer'
import { render, fireEvent, screen } from '@testing-library/react';
import { Button } from './Button'

describe('Button', () => {
  it('renders correctly with react-test-renderer', () => {
    const button = renderer.create(<Button />)
    expect(button.toJSON()).toMatchSnapshot()
  })
  it('changes the text after click', () => {
    const button = render(<Button />)
    fireEvent.click(button.getByText('ON'))
    expect(button.getByText(/OFF/i)).toBeTruthy()
    fireEvent.click(button.getByText('OFF'))
    expect(button.getByText(/ON/i)).toBeTruthy()
  })

  // it('debug screen', () => {
  //   const button = render(<Button />)
  //   screen.debug()
  // })
})
