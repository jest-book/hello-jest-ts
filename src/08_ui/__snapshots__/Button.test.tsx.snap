// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with react-test-renderer 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  ON
</button>
`;
